### ISSUE TITLE
Make plugin dependency section not dependent on maven version

### ISSUE THREAD

**Issue by YotillaAntoni** [exasol/project-keeper#599]
Example commit needed to fix the issue within a Virtual Snowflake Schema in https://github.com/exasol/snowflake-virtual-schema/pull/9/commits/768c0b52e8cb790785ba74167156b4ca4eb6b4d7

See also #585

**Comment by ckunki** [comment]
It also failed in VSBIGQ here:
https://github.com/exasol/bigquery-virtual-schema/actions/runs/11888127235/job/33122067977 
Locally I used mvn 3.8.8 but PK expected 

```
### Plugin Dependency Updates

* Updated `com.exasol:project-keeper-maven-plugin:4.3.3` to `4.4.0`
* Added `com.exasol:quality-summarizer-maven-plugin:0.2.0`
* Updated `io.github.zlika:reproducible-build-maven-plugin:0.16` to `0.17`
* Updated `org.apache.maven.plugins:maven-clean-plugin:3.2.0` to `3.4.0`
* Updated `org.apache.maven.plugins:maven-failsafe-plugin:3.2.5` to `3.5.1`
* Updated `org.apache.maven.plugins:maven-install-plugin:3.1.0` to `3.1.3`
* Updated `org.apache.maven.plugins:maven-jar-plugin:3.4.1` to `3.4.2`
* Updated `org.apache.maven.plugins:maven-resources-plugin:3.3.0` to `3.3.1`
* Updated `org.apache.maven.plugins:maven-site-plugin:3.12.1` to `3.9.1`
* Updated `org.apache.maven.plugins:maven-surefire-plugin:3.2.5` to `3.5.1`
* Updated `org.codehaus.mojo:versions-maven-plugin:2.16.2` to `2.17.1`
```

While in GitHub Runner PK expected 
```
### Plugin Dependency Updates

* Updated `com.exasol:project-keeper-maven-plugin:4.3.3` to `4.4.0`
* Added `com.exasol:quality-summarizer-maven-plugin:0.2.0`
* Updated `io.github.zlika:reproducible-build-maven-plugin:0.16` to `0.17`
* Updated `org.apache.maven.plugins:maven-clean-plugin:2.5` to `3.4.0`
* Updated `org.apache.maven.plugins:maven-failsafe-plugin:3.2.5` to `3.5.1`
* Updated `org.apache.maven.plugins:maven-install-plugin:2.4` to `3.1.3`
* Updated `org.apache.maven.plugins:maven-jar-plugin:3.4.1` to `3.4.2`
* Updated `org.apache.maven.plugins:maven-resources-plugin:2.6` to `3.3.1`
* Updated `org.apache.maven.plugins:maven-site-plugin:3.3` to `3.9.1`
* Updated `org.apache.maven.plugins:maven-surefire-plugin:3.2.5` to `3.5.1`
* Updated `org.codehaus.mojo:versions-maven-plugin:2.16.2` to `2.17.1`
```

**Comment by kaklakariada** [comment]
This can still happen when
* a project uses PK < 4.4.0
* and you upgrade to PK >= 4.4.0
* and you locally use a newer Maven version than GitHub action (e.g. 3.9.9)
PK 4.4.0 pins the version of all mentioned plugins. Once a project is released with PK 4.4.0 the Maven version is not relevant any more.
